src/overlays/actors/ovl_Obj_Kibako/z_obj_kibako.c

```diff
@@ -20,42 +39,375 @@ const ActorInit Obj_Kibako_InitVars = {
+}
+
+void ObjKibako_WaterBreak(ObjKibako* this, GlobalContext* globalCtx) {
+    s16 angle;
+    s32 i;
+    Vec3f* worldPos = &this->actor.world.pos;
+    Vec3f pos;
+    Vec3f velocity;
+
+    pos.y = worldPos->y + this->actor.yDistToWater;
+    for (angle = 0, i = 0; i < 5; i++, angle += 0x3333) {
+        pos.x = (Math_SinS(((s32)(Rand_ZeroOne() * 6000.0f)) + angle) * 15.0f) + worldPos->x;
+        pos.z = (Math_CosS(((s32)(Rand_ZeroOne() * 6000.0f)) + angle) * 15.0f) + worldPos->z;
+        EffectSsGSplash_Spawn(globalCtx, &pos, NULL, NULL, 0, 350);
+    }
+    pos.x = worldPos->x;
+    pos.z = worldPos->z;
+    EffectSsGRipple_Spawn(globalCtx, &pos, 200, 600, 0);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Kibako_0x809261B0/func_80926224.asm")
+    for (i = 0, angle = 0; i < 12; i++, angle += 0x4E20) {
+        f32 sn = Math_SinS(angle);
+        f32 cs = Math_CosS(angle);
+        f32 temp_rand;
+        s16 phi_s0;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Kibako_0x809261B0/func_809262BC.asm")
+        pos.x = sn * 16.0f;
+        pos.y = (Rand_ZeroOne() * 5.0f) + 2.0f;
+        pos.z = cs * 16.0f;
+        velocity.x = pos.x * 0.18f;
+        velocity.y = (Rand_ZeroOne() * 4.0f) + 2.0f;
+        velocity.z = pos.z * 0.18f;
+        pos.x += worldPos->x;
+        pos.y += worldPos->y;
+        pos.z += worldPos->z;
+        temp_rand = Rand_ZeroOne();
+        phi_s0 = (temp_rand < 0.2f) ? 0x40 : 0x20;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Kibako_0x809261B0/func_80926318.asm")
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, worldPos, -180, phi_s0, 50, 5, 0,
+                             (Rand_ZeroOne() * 35.0f) + 10.0f, 0, 0, 70, -1,
+                             sObjectBankTable[(this->actor.params >> 0xF) & 1],
+                             sParticleDisplayLists[(this->actor.params >> 0xF) & 1]);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Kibako_0x809261B0/func_80926394.asm")
+void func_80926B40(ObjKibako* this) {
+    this->actionFunc = func_80926B54;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Kibako_0x809261B0/ObjKibako_Init.asm")
+void func_80926B54(ObjKibako* this, GlobalContext* globalCtx) {
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    func_800B78B8(globalCtx, &this->actor, 18.0f, 15.0f, 0.0f, 0x45);
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->bankIndex) != 0) {
+        this->actor.draw = ObjKibako_Draw;
+        this->actor.objBankIndex = this->bankIndex;
+        ObjKibako_SetupIdle(this);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Kibako_0x809261B0/ObjKibako_Destroy.asm")
+void ObjKibako_SetupIdle(ObjKibako* this) {
+    this->actionFunc = ObjKibako_Idle;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Kibako_0x809261B0/func_8092650C.asm")
+void ObjKibako_Idle(ObjKibako* this, GlobalContext* globalCtx) {
```

Comment:
We call idle "Wait" usually

Reply:
This is what it's called in the OoT repo.